dataline-workers/src/main/java/io/dataline/workers/singer/SingerSyncWorker.java

```diff
@@ -68,16 +70,23 @@ public DefaultSyncWorker(TapFactory<SingerMessage> singerTapFactory,
 
     final SingerMessageTracker singerMessageTracker = new SingerMessageTracker();
 
-    try (Stream<SingerMessage> tap = singerTapFactory.create(tapConfig, jobRoot);
-        CloseableConsumer<SingerMessage> consumer = singerTargetFactory.create(targetConfig, jobRoot)) {
+    try {
+      singerTarget.start(targetConfig, jobRoot);
```

Comment:
this is pushing a bunch of problems that were caught by the compiler into things that can only be caught at runtime. specifically it's adding more statefulness to the object with this start concept why do we prefer this?

Reply:
discussed offline: there were issues around when streams are being closed, and so at this early stage empirically it was easier to debug and make sure our behavior was a correct by explicitly handling the lifecycle in one class without try-resources, etc.